File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.py
def solution(num):
    cnt = 0
    while True:
        if num == 1:
            break
        elif num % 2 == 0:
            num = num // 2
            cnt += 1
        else:
            num = num * 3 + 1
            cnt += 1
        
        if num == 1:
            break
        
        if cnt >= 500:
            return -1
            break
            
    return cnt

File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.java
class Solution {
    public int solution(long num) {
        int cnt = 0;
        
        while(num != 1) {
            if (num % 2 == 0) {
                num = num / 2;
            } else {
                num = num * 3 + 1;
            }
            cnt++;
            
            if (cnt >= 500) {
                cnt = -1;
                break;
            }
        }
        
        return cnt;
    }
}